xmtp_mls/src/identity.rs

```diff
@@ -137,13 +137,14 @@ impl Identity {
             )?
             .into();
             let credential =
-                OpenMlsCredential::new(credential_proto.encode_to_vec(), CredentialType::Basic)?;
+            OpenMlsCredential::new(CredentialType::Basic, credential_proto.encode_to_vec());
             self.set_credential(credential)?;
         }
 
         // Register the installation with the server
         let kp = self.new_key_package(provider)?;
-        let kp_bytes = kp.tls_serialize_detached()?;
+        let kp_bytes = kp.tls_serialize_detached()
+            .map_err(|e| IdentityError::TlsSerialization(e))?;
```

Comment:
Same as above for the `#[from]`. You should be able to automatically convert these